lib/elixir_console_web/live/console_live/sidebar_component.ex

```diff
@@ -6,4 +6,7 @@ defmodule ElixirConsoleWeb.ConsoleLive.SidebarComponent do
   """
 
   use Phoenix.LiveComponent
+
+  def variable_name({name, _extra}), do: name
```

Comment:
Gotcha...

Quickly checking the same expressions on Elixir 1.9.1 give the same result, so maybe that's not the cause or at least not the only reason...?

I can dig deeper to see if I can better understand what's happening behind the scenes.

Reply:
Yeah that's not the cause, that's just why the nils appear.
The cause is this:
```
Interactive Elixir (1.10.4)
iex(2)> Code.eval_quoted(Code.string_to_quoted!("nil"), [a: 1], [])
{nil, [{{:a, nil}, 1}]}
```
vs
```
Interactive Elixir (1.9.4)
iex(1)> Code.eval_quoted(Code.string_to_quoted!("nil"), [a: 1], [])
{nil, [a: 1]}
```

Reply:
Indeed.

It seems it's happening for every atom, and not for non atoms.

```
iex(6)> Code.eval_quoted(Code.string_to_quoted!("nil"), [a: 1], [])                         
{nil, [{{:a, nil}, 1}]}
iex(7)> Code.eval_quoted(Code.string_to_quoted!("false"), [a: 1], [])
{false, [{{:a, nil}, 1}]}
iex(8)> Code.eval_quoted(Code.string_to_quoted!("Kernel"), [a: 1], [])
{Kernel, [{{:a, nil}, 1}]}
iex(9)> Code.eval_quoted(Code.string_to_quoted!("1"), [a: 1], [])     
{1, [a: 1]}
```

Reply:
According to lines `109` to `112` in https://github.com/elixir-lang/elixir/compare/v1.9.4..v1.10.0#diff-a8ad23bd9671e17af264b79a023bb4ad5669dcb67a451f3df466c66cbc09e9b2 it seems there was an intentional expansion of what the binding type could be, from just a keyword list to possibly a list with tuples as keys.

So, despite maybe something strange with this particular `nil` case, according to the docs it makes sense to start expecting different both types of var binding keys (atoms and tuple) anyway?

Reply:
Good finding. So that's what we're doing right? Or do you propose other changes?

Reply:
Yep, doing this is fine.

Wondering whether a live view component is the best place to do this "normalization" though.

Maybe this is something better suited as a responsibility for the `Sandbox` to "normalize" or something closer to the business logic instead given the bindings are passed around a lot apparently.

Reply:
In fact, just realized the autocomplete is also broken after typing "a = 1; nil` also... because of https://github.com/wyeworks/elixir_console/blob/a3a3862bb26fd13d99e0a9fa34fcd73d75faa7de/lib/elixir_console/autocomplete.ex#L27-L31

Reply:
> Yeah that's not the cause, that's just why the nils appear.
> The cause is this:
> 
> ```
> Interactive Elixir (1.10.4)
> iex(2)> Code.eval_quoted(Code.string_to_quoted!("nil"), [a: 1], [])
> {nil, [{{:a, nil}, 1}]}
> ```
> 
> vs
> 
> ```
> Interactive Elixir (1.9.4)
> iex(1)> Code.eval_quoted(Code.string_to_quoted!("nil"), [a: 1], [])
> {nil, [a: 1]}
> ```

Apparently fixed now in latest Elixir v1.12.3 :smile: 

https://github.com/elixir-lang/elixir/commit/8d5c07c1a4c9f770e731aee1a946537cc9d1be5e

```
Interactive Elixir (1.12.3) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)> Code.eval_quoted(Code.string_to_quoted!("nil"), [a: 1], [])
{nil, [a: 1]} 
```